### PR TITLE
Support markdown code fences

### PR DESCRIPTION
**Pull request by gaborcsardi**
This is on top of #1024. It uses knitr to evaluate `` ```{r} `` markdown code fences and then insert the knitr output into the manual.

Features and notes
* Fences and inline code use the same evaluation environment.
* There is one evaluation environment per ~~markdown tag~~ roxygen block.
* ~~Plots are not yet supported.~~ Fixed them.
* It is possible to specify knitr chunk options.

Tests and docs are ~~coming~~ done.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
@hadley Added some tests, docs, and fixed plots. I think this is ready for review. Thanks!

**Comment by hadley** [comment]
I'll review this one once #1024 is finished and merged

**Comment by gaborcsardi** [comment]
@hadley FYI, I updated this after the changes in #1024.

File: vignettes/rd-formatting.Rmd [hunk 1]
```diff
@@ -263,6 +267,115 @@ Markdown syntax for inline images works. The image files must be in the `man/fig
...
+```r
+#' ```{r iris-pairs-plot}
+#' pairs(iris[1:4], main = "Anderson's Iris Data -- 3 species",
+#'   pch = 21, bg = c("red", "green3", "blue")[unclass(iris$Species)])
+#' ```
+```
+
+Note that code block support is currently experimental, and somewhat limited.
+Some of the known limitations:
+  * Because the code blocks are evaluated individually, they cannot refer to each other.
```
Comment (hadley) [inline vignettes/rd-formatting.Rmd]: Run on `document()` so need to be wary about performance?

File: vignettes/rd-formatting.Rmd [hunk 2]
```diff
@@ -263,6 +267,115 @@ Markdown syntax for inline images works. The image files must be in the `man/fig
...
+Code blocks support knitr chunk options, e.g. to keep the output of several expressions together, you can specify `results= "hold"`:
+
+```r
+#' ```{r results = "hold"}
+#' names(mtcars)
+#' nrow(mtcars)
+#' ```
+```
+
+Plots will create `.png` files in the `man/figures` directory. The file names are created from the chunk names:
```
Comment (hadley) [inline vignettes/rd-formatting.Rmd]: Probably worth a caveat about creating large packages?

**Comment by gaborcsardi** [comment]
Rebased and added the two caveats. Will merge once the CIs are done.

**Comment by gaborcsardi** [comment]
The CI failures seem unrelated.

**Comment by gaborcsardi** [comment]
@hadley Thanks, merged!

**Pull request merged**